### PR TITLE
feat(edgeless): edgless element drag snap

### LINKED ISSUE #2440: Support snap in whiteboard
This would allow auto align dragged item to other ones on whiteboard.

(from Azure)

### PR DESCRIPTION
**Pull request by regischen**
Closes #2440 
It is the design draft of edgeless element snap.
there are two things necessarily to be viewd. @doodlewind 
1. the API of `EdgelessSnapManager` and how it can be used.
2. I add mechanism to allow renderer to render non-phaserElement.

### CODE REVIEW

File: packages/blocks/src/page-block/edgeless/snap-manager.ts [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+                new Point(xs[3], y),
+                new Point(xs[4], y),
+              ];
+            }
+          };
+          if (lb.horizontalDistance(rb) > bound.w) {
+            _centerX = (lb.maxX + rb.minX) / 2;
+            updateDif();
+          }
+          /** 左边对齐 */
```
Comment (doodlewind) [inline packages/blocks/src/page-block/edgeless/snap-manager.ts:256]: Don't use Chinese comment

Comment (regischen) [reply packages/blocks/src/page-block/edgeless/snap-manager.ts:259]: ookay, I will notice in the following

**Review by doodlewind — CHANGES_REQUESTED** [review]
1. Some trivial refactoring could be done with the help of GPT.
2. The newly added `Renderable` concept looks a bit arbitrary and is something that doesn't fit perfectly into our CRDT-driven system in phasor. I need to think about the best practice  here.

**Review by doodlewind — APPROVED** [review]
LGTM, thanks for adding this critical functionality!

**Comment by regischen** [comment]
> LGTM, thanks for adding this critical functionality!
Also thanks for greatly improving quality of these codes!
now it fits more naturally into the project!

**Comment by regischen** [comment]
> LGTM, thanks for adding this critical functionality!

Also thanks for greatly improving quality of these codes!
now it fits more naturally into the project!

**Pull request merged**